text/0000-kafka-topics-per-tenant.md

```diff
@@ -0,0 +1,100 @@
+
+- Start Date: 2022-08-12
+- RFC PR: (leave this empty)
+- FOLIO Issue: (leave this empty)
+
+# Cease Creating Kafka Topics For Each Tenant
+
+## Summary
+
+The purpose of this RFC is to lay out the reasoning for creating Kafka topics using a different scheme other than what is currently employed; creating the same topic for each tenant. The recommendation is to create single topic for each application concern. This will reduce the partition count needed for normal operations.
+
+## Motivation
+FOLIO should cease creating Kafka topics for each tenant installed. This has scaling & cost implications.
```

Comment:
I added some words in the summary. Let me know if that is suitable, my expectation was not a someone with zero knowledge of Kafka to read and understand this RFC.

Reply:
> I added some words in the summary. Let me know if that is suitable

Thank you for adding additional information to the summary. I appreciate you expanding upon the relationship between topics and partitions.

I think a challenge I'm having is separating the why, what and how of this proposal from each other.

Here is my understanding so far:

why? Partitions are how cloud providers charge for Kafka. The more partitions a system uses, the more expensive it is to operate. We want to reduce the operating 
what? In order to reduce the operating cost of the system, we need to reduce the number of partitions in the system
how? Reducing the number of topics will reduce the number of partitions (because FOLIO uses a fixed number of partitions per topic). This can be done by combining per-tenant topics into a single topic.

> my expectation was not a someone with zero knowledge of Kafka to read and understand this RFC.

I think it's worth keeping in mind that folks on the TC, and the broader audience (as RFCs are publicly reviewed and act as documentation), may have varying degrees of knowledge on Kafka. Folks may understand the basics of how it is used in FOLIO, yet not understand how concepts like topics and partitions relate to each other.

Reply:
Your why	, what and how look good. A small correction is that FOLIO does not always dictate the number of partitions. Some modules in FOLIO demand use of a topic (consuming/producing) without any knowledge of the topic being created in the first place. So the auto-create feature of Kafka is used which create topic when, for example, a module needs to publish to a non-existent topic. The default partition number of a topic created via auto-create is one. All this is to say that it is not a rule in that every topic has a partition count set by FOLIO.

Reply:
> All this is to say that it is not a rule in that every topic has a partition count set by FOLIO.

I had no idea about this. I thought that had been established as a standard by the EPAM folks.